Documentation/Communicational Patterns/2. Tooling/2.5. Environment.md

```diff
@@ -0,0 +1,20 @@
+![document type: vision](https://themomax.github.io/resources/markdown-labels/document_type_vision.svg)
+
+# Environment
+
+The `@Environment` property wrapper provides different functionality depending on the `wrappedValue`. For more detail on how the `wrappedValue` is provided to the environment and what values are provided by the Apodini framework refer to [PropertyWrapper -> RequestAndEnvironment](../../PropertyWrapper/RequestAndEnvironment.md).
+
+## ObservableObject
+
+If the `wrappedValue` conforms to `ObservableObject`, the `@Environment` behaves exactly the same as [`@ObservedObject`](./2.4.%20ObservedObject.md).
+
+In any other case `@Environment` has no influence on the `Handler`'s lifetime or control flow.
+
+## Connection
+
+Apodini provides a `Connection` through the environment. This object can be accessed using `@Environment(\.connection)`. The `Connection` provides useful protocol-agnostic information about the current state of the communication. The `Connection` could provide a `start: Date` so the service could e.g. close connections with an timeout error.
+
+The most important property of `Connection` is `state: ConnectionState`. This state can be `.open` or `.end`, where `.end` signalizes that the protocol expects the connection to be closed now. The `Handler` will be destructed after the next evaluation that results in a `.send(Request)`. Usage of this special environment object may cause exporters to utilize a client-side stream for implementation of the respective endpoint.
```

Comment:
I would argue that the connection is closed independent if the next response from the service is a `.send(Request)` or a `.nothing`. I would suggest that the `.end` `Connection` property indicates that the client has the intention of terminating the connection and this provides a single last chance to respond before the connection is eventually closed, otherwise we would have zombie connections around that simply refuse to send a response back to the client and if they eventually do the client is long disconnected.

Reply:
This is indeed a good point I didn't think about in too much detail. I adapted the paragraph accordingly.